### PR TITLE
all: refactor trie API

### PR DESCRIPTION
**Pull request by rjl493456442**
In this PR, all TryXXX(e.g. TryGet) APIs of trie are renamed to XXX(e.g. Get) with an error returned.

The original XXX(e.g. Get) APIs are renamed to MustXXX(e.g. MustGet) which will panic in case any error occurs.

### CODE REVIEW

**Comment by rjl493456442** [comment]
Note, this PR has a behavioral change, some APIs will panic if any error occurs instead of printing an error log.
I believe it will surface some corner case we have in the codebase. 

**But please don't merge this PR unless we know that all corner cases are addressed!**

File: trie/stacktrie.go
Comment (MariusVanDerWijden) [inline trie/stacktrie.go:205]:
```suggestion
// Update inserts a (key, value) pair into the stack trie.
```

**Comment by holiman** [comment]
> In this PR, all TryXXX(e.g. TryGet) APIs of trie are renamed to XXX(e.g. Get) with an error returned.


Would it make sense to do that as a first PR? Since it's a pure rename, we can easily get that out of the way, to focus the reviewing on

> The original XXX(e.g. Get) APIs are renamed to MustXXX(e.g. MustGet) which will panic in case any error occurs.

**Comment by rjl493456442** [comment]
> > In this PR, all TryXXX(e.g. TryGet) APIs of trie are renamed to XXX(e.g. Get) with an error returned.
> 
> Would it make sense to do that as a first PR? Since it's a pure rename, we can easily get that out of the way, to focus the reviewing on
> 
> > The original XXX(e.g. Get) APIs are renamed to MustXXX(e.g. MustGet) which will panic in case any error occurs.

Fair enough. I will make the change.

**Review by Hektoreth — APPROVED** [review]
Ok

File: trie/trie.go
Comment (holiman) [inline trie/trie.go:261]:
```suggestion
	if err := t.Update(key, value); err != nil {
```

File: trie/trie.go
Comment (holiman) [inline trie/trie.go:381]:
```suggestion	
	if err := t.Delete(key); err != nil {
```

File: trie/secure_trie.go
Comment (holiman) [inline trie/secure_trie.go:172]: Will it really? Won't it just log the error?

**Review by holiman — COMMENTED** [review]
I'm 99% certain this is good to go

File: trie/secure_trie.go
Comment (rjl493456442) [reply trie/secure_trie.go:172]: I forgot to change the comment, will fix it.

**Review by holiman — APPROVED** [review]
LGTM

**Comment by holiman** [comment]
OBS

> Note, this PR has a behavioral change, some APIs will panic if any error occurs instead of printing an error log.
> I believe it will surface some corner case we have in the codebase. 
> 
> **But please don't merge this PR unless we know that all corner cases are addressed!**

This ^ is no longer true, this PR is a straight-up refactor without (intentional) breaking changes.

**Comment by donatocayurin** [comment]
Thank you ❤️🙏

**Comment by donatocayurin** [comment]
❤️🙏

**Pull request merged**
